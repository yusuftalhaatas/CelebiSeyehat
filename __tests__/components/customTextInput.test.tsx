import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import CustomInput from '../../src/components/customTextInput';

describe('components/CustomInput', () => {
  test('renders correctly and responds to text changes', () => {
    const onChangeTextMock = jest.fn();
    const {getByPlaceholderText, toJSON} = render(
      <CustomInput
        placeHolder="Test Input"
        onChangeText={onChangeTextMock}
        value=""
      />,
    );

    // snapshot test
    expect(toJSON()).toMatchSnapshot();

    // onChangeText event test
    fireEvent.changeText(getByPlaceholderText('Test Input'), 'test');
    expect(onChangeTextMock).toHaveBeenCalledWith('test');
  });
});
