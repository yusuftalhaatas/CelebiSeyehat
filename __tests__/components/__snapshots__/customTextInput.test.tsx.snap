// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/CustomInput renders correctly and responds to text changes 1`] = `
<View
  style={
    {
      "borderRadius": 10,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "flex-start",
      "marginHorizontal": 30,
      "marginVertical": 10,
      "paddingLeft": 10,
    }
  }
>
  <TextInput
    autoCapitalize="none"
    onChangeText={[MockFunction]}
    placeholder="Test Input"
    value=""
  />
</View>
`;
